src/org/ods/orchestration/usecase/JiraUseCase.groovy

```diff
@@ -464,4 +474,131 @@ class JiraUseCase {
+    }
+
+    String createOrUpdateSecurityVulnerabilityIssue(String vulnerabilityName, String jiraComponentId,
+                                                    String description) {
+        if (!jira) {
+            throw new JiraNotPresentException("JiraUseCase not present, cannot create security vulnerability issue.")
+        }
+
+        def issueSummary =  SECURITY_VULNERABILITY_ISSUE_SUMMARY.replace(VULNERABILITY_NAME_PLACEHOLDER,
+            vulnerabilityName)
+
+        def fixVersion = null
+        if (project.isVersioningEnabled) {
+            fixVersion = project.getVersionName()
+        }
+        def fullJiraComponentName = JIRA_COMPONENT_TECHNOLOGY_PREFIX + jiraComponentId
+
+        List securityVulnerabilityIssues = loadJiraSecurityVulnerabilityIssues(issueSummary,
+            fixVersion, fullJiraComponentName, project.jiraProjectKey)
+        if (securityVulnerabilityIssues?.size() >= 1) { // Transition the issue to "TO DO" state
+            transitionIssueToToDo(securityVulnerabilityIssues.get(0).id)
+            return (securityVulnerabilityIssues.get(0) as Map)?.key
+        } else { // Create the issue
+            return (createIssueTypeSecurityVulnerability(fixVersion: fixVersion, component: fullJiraComponentName,
+                priority: SECURITY_VULNERABILITY_ISSUE_PRIORITY, projectKey: project.jiraProjectKey,
+                summary: issueSummary, description: description)
+                as Map)?.key
+        }
+    }
+
+    String buildSecurityVulnerabilityIssueDescription(Map vulnerability, String gitUrl, String gitBranch,
+                                                      String repoName, String nexusReportLink) {
+        StringBuilder message = new StringBuilder()
+        message.append("\nAqua security scan detected the remotely exploitable critical " +
+            "vulnerability with name *${vulnerability.name as String}* in repository *[${repoName}|${gitUrl}]* " +
+            "in branch *${gitBranch}*." )
+        message.append("\n\n*Description:* " + vulnerability.description as String)
+        message.append("\n\n*Solution:* " + vulnerability.solution as String)
+
+        if (nexusReportLink != null) {
+            message.append("\n\nYou can find the complete security scan report *[here|${nexusReportLink}]*.")
+        }
+
+        return message.toString()
+    }
+
+    Map createIssueTypeSecurityVulnerability(Map args) {
+        return jira?.createIssue(fixVersion: args.fixVersion, component: args.component,
+            priority: args.priority, summary: args.summary, type: "Security Vulnerability",
+            projectKey: args.projectKey, description: args.description)
+    }
+
+    void transitionIssueToToDo(String issueId) {
+        int maxAttemps = 10;
+        while (maxAttemps-- > 0) {
+            Map response = jira?.getIssueStatusWithTransitions(issueId)
+            String issueStatus = response.status
+            Map possibleTransitionsByName = response.transitions.
```

Comment:
Obtaining the possible transitions can be moved after check if is a "to do". We don't need the tramsitions before that.

Reply:
We get the issue status with expanded transitions with that call so we need the call to get the status.

Reply:
Moved it after the transition collection.